src/Microsoft.ML.Data/Transforms/ValueMappingTransformer.cs

```diff
@@ -191,18 +192,15 @@ private static ValueGetter<VBuffer<ReadOnlyMemory<char>>> GetKeyValueGetter<TKey
                 // set of values. This is used for generating the metadata of
                 // the column.
                 HashSet<TValue> valueSet = new HashSet<TValue>();
-                HashSet<TKey> keySet = new HashSet<TKey>();
                 for (int i = 0; i < values.Count(); ++i)
                 {
                     var v = values.ElementAt(i);
```

Comment:
Can you replace it with foreach loop?
I found terrifying idea of fetching i-th element every time out of IEnumerable

Reply:
absolutely.

---
In reply to: [246975951](https://github.com/dotnet/machinelearning/pull/2098#discussion_r246975951) [](ancestors = 246975951)